### DISCUSSION TITLE
Best way to show one world without limits on boundaries

### DISCUSSION THREAD

**Discussion by DervoDev** [maplibre/maplibre-gl-js#5096] [General]
What is the best way to show a world map once, and where Russias eastern-most part is not cut of at the antimeridian? 
Ideally the map shows something like this: 
<img width="1047" alt="Screenshot 2024-11-22 at 13 05 34" src="https://github.com/user-attachments/assets/5bfe2d41-8400-48ce-8efe-123b67600037">
Only showing background color (blue) on the world-copies on each side, except for the eastern-most part of Russia that is replaced at 

I have done some attempts: 
Using setRenderWorldCopies(false) turns off the 3 world-copies at each side, and at the same time limiting the cameras horizontal max-bounds to -180 to 180 degrees. This leads to cutting off a portion of either north or south of the world in a ultra-widescreen setup. 
Creating a mask to cut of and shifting it over to the tile over just wraps it around.

**Comment by HarelM** [comment]
I think you can `setBounds`, can't you?

**Reply by DervoDev** [reply]
map.setMaxBounds() gets weird when attempting to set a max bounds of lower then -180 and higher then 180. It does not set correct max bounds when using 

Running the following code provides the following output: 
<img width="347" alt="image" src="https://github.com/user-attachments/assets/4ff6a0ec-5225-4e35-9b25-516cadf10e1e">

<img width="439" alt="image" src="https://github.com/user-attachments/assets/f872c9be-0aeb-4517-bbd8-9829509debdd">

Attempting to set `map.setRenderWorldCopies(false)` gives no difference.

**Reply by HarelM** [reply]
Hmm... I don't have other good ideas... I guess you'll need to dig into the code and see what can be done to solve this...
